src/unix/mediactrl_gstplayer.cpp

```diff
@@ -337,6 +375,18 @@ bool wxGStreamerMediaBackend::CreateControl(wxControl* ctrl, wxWindow* parent,
     m_video_renderer = gst_player_video_overlay_video_renderer_new(window_handle);
     m_player = gst_player_new(m_video_renderer, gst_player_g_main_context_signal_dispatcher_new(NULL));
 
+    wxDisplayInfo info = wxGetDisplayInfo();
+    if (info.type == wxDisplayWayland) {
+        // wayland needs a specific handler to pass display to gstreamer
+        GstBus *bus = gst_pipeline_get_bus(GST_PIPELINE(gst_player_get_pipeline(m_player)));
+        gst_bus_add_signal_watch(bus);
+        gst_bus_set_sync_handler(bus, bus_sync_handler, this, NULL);
+        gst_object_unref(bus);
```

Comment:
Sorry, this is not really related to this PR and shouldn't be done in it, but I just wanted to mention that if anybody is interested in copying `wxGtkObject` from `include/wx/gtk/private/object.h` to some `wxGstObject` and using it here, it would be welcome!